utils/hwloc/misc.h

```diff
@@ -325,6 +335,19 @@ hwloc_utils_enable_input_format(struct hwloc_topology *topology, unsigned long f
     err = hwloc_utils_enable_input_format(topology, flags, subdir, &sub_input_format, verbose, callname);
     if (!err)
       *input_format = sub_input_format;
+    else {
+      close(oldworkdir);
+      return err;
+    }
+
+    err = fchdir(oldworkdir);
+    if (err) {
+      perror("Restoring current working directory");
+      close(oldworkdir);
+      return EXIT_FAILURE;
+    }
+
+    close(oldworkdir);
```

Comment:
Not sure that's enough. The goal is to keep a reference on the archivemount'ed path during topology_load(), which happens after this code. So you'd likely need to move fchdir+close after load(). That's a bit ugly but I guess we'd need to make oldworkdir a global variable initialized to -1, and that variable would need to be added to all users of "input_format" macros :/